packages/tools/fetch-tool/src/fluidFetchMessages.ts

```diff
@@ -140,16 +187,24 @@ async function* saveOps(
     // Split into 100K ops
     const chunk = 100 * 1000;
 
+    let sequencedMessages: ISequencedDocumentMessage[] = [];
+
     // Figure out first file we want to write to
     let index = 0;
+    let curr: number = 1;
     if (files.length !== 0) {
-        index = (files.length - 1);
+        index = files.length - 1;
+        const name = filenameFromIndex(index);
+        const fileContent = fs.readFileSync(`${dir}/messages${name}.json`, { encoding: "utf-8" });
+        const messages: ISequencedDocumentMessage[] = JSON.parse(fileContent);
+        curr = messages[0].sequenceNumber;
     }
-    let curr = index * chunk + 1;
 
-    let sequencedMessages: ISequencedDocumentMessage[] = [];
     while (true) {
         const result: IteratorResult<ISequencedDocumentMessage[]> = await gen.next();
+        if (files.length === 0) {
```

Comment:
I don't think you need to do it inside while loop. Just do it above as "else" part of "if (files.length !== 0) {"

Reply:
so i'm putting it after `await gen.next()` in line 204, otherwise it wouldn't get firstAvailableDelta

Reply:
ok